src/app/shared/services/publish-role/publish-role.service.ts

```diff
@@ -115,25 +116,33 @@ export class PublishRoleService {
       .pipe(truthy());
   }
 
-  private syncClaimToDidDoc(element: {
+  private async syncClaimToDidDoc(element: {
     issuedToken: string;
     registrationTypes: RegistrationTypes[];
     claimType: string;
+    claimTypeVersion: string;
   }) {
+    const isRegisteredOnChain = await this.checkForNotSyncedOnChain(element);
+    const { notSyncedOnChain } = isRegisteredOnChain;
+    // If the element is already synced on chain, only off-chain registration is needed:
+    const registrationTypes = notSyncedOnChain
+      ? element.registrationTypes
+      : [RegistrationTypes.OffChain];
```

Comment:
Ah, thanks @dawidgil that makes sense. I moved this logic to publish-role service here. LMK if you think this makes better sense. If correct, I will update failing unit tests for toastr. Thank you!

Reply:
Cool. I just checked what's wrong with tests. I had to change async/await code into rxjs like and they are passing. If you want to play with rxjs here, then i can just push unit tests (added one locally that checks registrationTypes). If you want entire solution for this, i can push that as well :)